packages/teams-js/test/public/appNotification.spec.ts

```diff
@@ -0,0 +1,215 @@
+                     Our team at Microsoft is working diligently to ensure a smooth transition during this maintenance period. We understand the importance of your business operations and are committed to minimizing any disruptions. However, there may be temporary service interruptions during the upgrade process. We apologize for any inconvenience this may cause and appreciate your understanding.
+                     We recommend that you plan accordingly for this extended maintenance and notify your team about the possible impact on their work. It is essential to ensure all critical operations are completed before the maintenance window to avoid any potential data loss or service disruptions.
+                     Once the maintenance is complete, you can expect enhanced performance, improved stability, and better security measures. Our goal at Microsoft is to provide you with the best possible service, and these upgrades are a crucial step towards achieving that objective.
+                     If you have any questions or concerns regarding the upcoming maintenance or its impact on your business, please don't hesitate to contact our support team. We are here to assist you and address any queries you may have.
+                     Thank you for your cooperation and continued support as we work to deliver a more robust and secure service. We value your business and are committed to delivering exceptional experiences to all our users.
+
+                     Best regards,
+                     Microsoft`,
+            displayDurationInSeconds: 30,
+            notificationActionUrl: new URL('http://www.example.com'),
+          };
+
+          expect(() => appNotification.displayInAppNotification(displayNotificationParam)).toThrowError(
+            new Error(
+              `Invalid notification content length: Maximum content length ${maxContentLength}, content length supplied ${displayNotificationParam.content.length}`,
+            ),
+          );
+        });
+
+        it('displayInAppNotification rejects promise with Error when error received from host', async () => {
+          await utils.initializeWithContext(context);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: { appNotification: {} } });
+          const promise = appNotification.displayInAppNotification(displayNotificationParam);
+
+          const message = utils.findMessageByFunc('appNotification.displayNotification');
+          expect(message).not.toBeNull();
+          expect(message.args.length).toBe(1);
+          expect(message.args[0]).toEqual(displayNotificationParamForAppHost);
+
+          const callbackId = message.id;
+          utils.respondToFramelessMessage({
+            data: {
+              id: callbackId,
+              args: [{ errorCode: ErrorCode.INTERNAL_ERROR }],
+            },
+          } as DOMMessageEvent);
+          await expect(promise).rejects.toEqual({ errorCode: ErrorCode.INTERNAL_ERROR });
+        });
+      } else {
```

Comment:
>else [](http://example.com/codeflow?start=8&length=4)

What does this `else` statement do here?

Reply:
It's the corresponding else statement for if the current frame context is included in the list of allowedContexts 
    line 65)  if (allowedContexts.some((allowedContext) => allowedContext === context)) { ....